drivers/scheduler/k8s/k8s.go

```diff
@@ -1150,51 +1149,53 @@ func (k *K8s) destroyCoreObject(spec interface{}, opts map[string]bool, app *spe
 
 }
 
-func (k *K8s) createAutopilotObject(apParams *scheduler.AutopilotParameters) (*apapi.AutopilotRule, error) {
-	obj := &apapi.AutopilotRule{}
-	apRuleParams := apParams.AutopilotRuleParameters
-	obj.Name = apParams.Name
-	obj.Namespace = apParams.Namespace
-	obj.Labels = map[string]string{"creator": "torpedo"}
-	obj.Spec.PollInterval = apParams.PollInterval
-	obj.Spec.ActionsCoolDownPeriod = apRuleParams.ActionsCoolDownPeriod
-	obj.Spec.Selector.LabelSelector.MatchLabels = apRuleParams.MatchLabels //map[string]string{"name": "pgbench-data"}
-	obj.Spec.NamespaceSelector.LabelSelector.MatchLabels = map[string]string{"creator": "torpedo"}
-	for _, ruleExpression := range apRuleParams.RuleConditionExpressions {
-		exprVolumeUsage := &apapi.LabelSelectorRequirement{
-			Key:      ruleExpression.Key,
-			Operator: apapi.LabelSelectorOperator(ruleExpression.Operator),
-			Values:   ruleExpression.Values,
-		}
-		obj.Spec.Conditions.Expressions = append(obj.Spec.Conditions.Expressions, exprVolumeUsage)
+func (k *K8s) addAppSpecEnvironmentVariables(containerEnvVars []v1.EnvVar, appEnvVars []scheduler.EnvVar) []v1.EnvVar {
+
+	var tmpEnvMapVars = make(map[string]string)
+	var envVars []v1.EnvVar
+	for _, cEnvVar := range containerEnvVars {
+		tmpEnvMapVars[cEnvVar.Name] = cEnvVar.Value
 	}
-	for _, ruleAction := range apRuleParams.RuleActions {
-		actions := &apapi.RuleAction{
-			Name:   ruleAction.Name,
-			Params: ruleAction.Params,
-		}
-		obj.Spec.Actions = append(obj.Spec.Actions, actions)
+	for _, appEnvVar := range appEnvVars {
+		tmpEnvMapVars[appEnvVar.Name] = appEnvVar.Value
+	}
+	for key, value := range tmpEnvMapVars {
+		envVars = append(envVars, v1.EnvVar{
+			Name:  key,
+			Value: value,
+		})
```

Comment:
you should be able to reduce one for loop here.  Skip the tmpEnvMapVars and directly append to the envVars in the first 2 loops.